File: frontend/src/api/doc.ts
import request from './request'
import blobClient from './blobClient'

/** 文档类型（与后端 DocService.DOC_TYPE_* 一致） */
export const DOC_TYPES = ['MARKDOWN', 'TEXT', 'HTML', 'PDF', 'IMAGE'] as const

export type DocType = (typeof DOC_TYPES)[number]

export interface DocItem {
  id: number
  title: string
  templateId?: number
  templateName?: string
  content: string
  docType?: DocType
  status: string
  createdBy?: string
  createdTime?: string
  updatedTime?: string
  version?: number
  approvedBy?: string
  rejectReason?: string
  /** 最近一次「上传到 IFS」的路径（发布/手动上传后回填） */
  ifsPath?: string
}

export interface TemplateItem {
  id: number
  name: string
  category: string
  content?: string
  docType?: DocType
  createdBy?: string
  createdTime?: string
  updatedTime?: string
}

export interface DocVersion {
  id: number
  version: number
  title: string
  content: string
  createdTime: string
  operator?: string
}

export const listDocs = (params: {
  current?: number
  size?: number
  keyword?: string
  status?: string
  deleted?: number
} = {}): Promise<{ total: number; records: DocItem[] }> => request.get('/docs', { params })

export const createDoc = (data: { title: string; templateId?: number; content: string; docType?: DocType }) =>
  request.post('/docs', data)

export const updateDoc = (id: number, data: { title: string; templateId?: number; content: string; docType?: DocType; ifsPath?: string }) =>
  request.put(`/docs/${id}`, data)

export const deleteDoc = (id: number) => request.delete(`/docs/${id}`)

export const restoreDoc = (id: number) => request.post(`/docs/${id}/restore`)

export const purgeDoc = (id: number) => request.delete(`/docs/${id}/purge`)

export const submitDoc = (id: number) => request.post(`/docs/${id}/submit`)

export const approveDoc = (id: number) => request.post(`/docs/${id}/approve`)

export const rejectDoc = (id: number, reason: string) =>
  request.post(`/docs/${id}/reject`, null, { params: { reason } })

export const docVersions = (id: number): Promise<DocVersion[]> => request.get(`/docs/${id}/versions`)

export const rollbackDoc = (id: number, version: number) =>
  request.post(`/docs/${id}/rollback/${version}`)

/**
 * 文档 IFS 发布文件（V49 文件型预览）：走 blob 实例携带鉴权头，返回原始 Blob。
 * PDF/图片 等二进制类型经它按 Content-Type 展示。
 */
export const docFile = async (id: number): Promise<Blob> => {
  const res = await blobClient.get<Blob>(`/docs/${id}/file`)
  return res.data
}

export const listTemplates = (category?: string): Promise<TemplateItem[]> =>
  request.get('/doc-templates', { params: { category } })

export const createTemplate = (data: { name: string; category: string; content: string; docType?: DocType }) =>
  request.post('/doc-templates', data)

export const updateTemplate = (id: number, data: { name: string; category: string; content: string; docType?: DocType }) =>
  request.put(`/doc-templates/${id}`, data)

export const deleteTemplate = (id: number) => request.delete(`/doc-templates/${id}`)

File: backend/rxas400adm-as400/src/main/java/com/rxas400adm/as400/entity/Doc.java
package com.rxas400adm.as400.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 运维文档（rx_doc，3.9）：模板起草 + 版本管理 + 审批状态机。
 * 状态：DRAFT → PENDING（提交审批）→ PUBLISHED（通过）/ REJECTED（驳回）。
 */
@Data
@TableName("rx_doc")
public class Doc {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long templateId;

    private String title;

    private String content;

    /** 文档类型：MARKDOWN / TEXT / HTML（正文渲染）/ PDF / IMAGE（IFS 文件渲染） */
    private String docType;

    private Integer version;

    /** DRAFT / PENDING / PUBLISHED / REJECTED */
    private String status;

    private String rejectReason;

    /** 最近一次上传到 IFS 的目标路径（文档管理「上传到 IFS」） */
    private String ifsPath;

    /** 逻辑删除标记：0=正常 1=已删除（删除文档=标记不物理删，保留审批/版本历史） */
    private Integer deleted;

    /** 逻辑删除时间（已删除视图展示用） */
    private LocalDateTime deletedTime;

    private String createdBy;

    private String updatedBy;

    private String approvedBy;

    private LocalDateTime approvedTime;

    private LocalDateTime createdTime;

    private LocalDateTime updatedTime;

    /** 模板名称（列表展示用，非表字段） */
    @TableField(exist = false)
    private String templateName;
}
